src/ParcelRegistry.Migrator.Parcel/Infrastructure/StreamMigrator.cs

```diff
@@ -180,33 +185,54 @@ await Policy
+            {
+                var migrateParcel = legacyParcelAggregate.CreateMigrateCommand(addressIds
+                        .Where(x => x.isSuccess)
+                        .Select(x => x.addressPersistentLocalId)
+                        .ToList(),
+                    new ExtendedWkbGeometry(grbParcel.Geometry.ToBinary()));
 
-            var markMigrated = new MarkParcelAsMigrated(
-                migrateParcel.OldParcelId,
-                migrateParcel.Provenance);
+                var markMigrated = new MarkParcelAsMigrated(
+                    migrateParcel.OldParcelId,
+                    migrateParcel.Provenance);
 
-            await DispatchCommand(markMigrated, ct);
-            await DispatchCommand(migrateParcel, ct);
+                await DispatchCommand(markMigrated, ct);
+                await DispatchCommand(migrateParcel, ct);
 
-            await _processedIdsTable.Add(internalId);
-            processedItems.Add(internalId);
+                await _processedIdsTable.Add(internalId);
+                processedItems.Add(internalId);
 
-            await using var backOfficeContext = streamLifetimeScope.Resolve<BackOfficeContext>();
-            foreach (var addressPersistentLocalId in migrateParcel.AddressPersistentLocalIds)
-            {
-                await backOfficeContext
-                    .ParcelAddressRelations.AddAsync(
-                        new ParcelAddressRelation(
-                            migrateParcel.NewParcelId,
-                            addressPersistentLocalId), ct);
+                await using var backOfficeContext = streamLifetimeScope.Resolve<BackOfficeContext>();
+                foreach (var addressPersistentLocalId in migrateParcel.AddressPersistentLocalIds)
+                {
+                    await backOfficeContext
+                        .ParcelAddressRelations.AddAsync(
+                            new ParcelAddressRelation(
+                                migrateParcel.NewParcelId,
+                                addressPersistentLocalId), ct);
+                }
+
+                await backOfficeContext.SaveChangesAsync(ct);
             }
+            else
+            {
+                _logger.LogWarning($"Parcel geometry not found for '{aggregateId}', retiring parcel.");
+
+                var provenance = new Provenance(
+                    SystemClock.Instance.GetCurrentInstant(),
+                    Application.ParcelRegistry,
+                    new Reason("Migrate Parcel aggregate."),
+                    new Operator("Parcel Registry"),
+                    Modification.Insert,
+                    Organisation.DigitaalVlaanderen);
 
-            await backOfficeContext.SaveChangesAsync(ct);
+                await DispatchCommand(new RetireParcel(parcelId, provenance), ct);
+                await DispatchCommand(new MarkParcelAsMigrated(parcelId, provenance), ct); //TODO: do we mark it or not?
```

Comment:
So we are not migrating retired parcels?

Reply:
After thinking about it more, it does makes sense. Cause when we will process the delta's when an explicit delete is captured we will also retire the parcel. Only because it has no geometry we can't migrate it.